### ISSUE TITLE
Resume doesn't work properly with parametrized test

### ISSUE THREAD

**Issue by sergei-dyshel** [getslash/slash#476]
``` python
import slash

@slash.parameters.iterate(param=[1,2,3])
def test_multi(param):
  if param == 3:
    raise Exception('bad param')
```

```
$ slash run .                     
3 tests collected   
./test_multi.py ..E  FAIL
============================================= Session Summary ==============================================
== #3: ./test_multi.py:test_multi(param=3) =================================================================
 - 1/1 E (2016-09-22 15:28:59 +03:00):   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 /usr/lib/python2.7/site-packages/slash/core/fixtures/parameters.py:30
>                return func(*args, **kwargs)
 /home/sergei/temp/slash-param/test_multi.py:6
>    raise Exception('bad param')
E    Exception: bad param
========== Session ended. 2 successful, 0 skipped, 0 failures, 1 errors. Total duration: 00:00:00 ==========

$ slash resume            
0 tests collected   
========== Session ended. 0 successful, 0 skipped, 0 failures, 0 errors. Total duration: 00:00:00 ==========
```

This doesn't happen with simple tests (without parametritization).

I tracked execution up to `loader.py:Loader._address_in_file_matches()` which seems to work incorectly and overall looks a bit hacky :smile:

**Comment by vmalloc** [comment]
Agreed. We need to overhaul the whole resume mechanism. It also currently doesn't handle parallel runs very well. Adding to the list.

**Comment by vmalloc** [comment]
Fixed in #531
